Failing to compile current master with LLVM 6.0 after IR changes

Just to keep track of the current situation of Julia master regarding LLVM 6.0.

With todays master, I am experiencing the following error (after the changes in the IR on the last days)

```
/home/lenz/julia/src/llvm-late-gc-lowering.cpp: In member function ‘bool LateLowerGCFrame::CleanupIR(llvm::Function&, State*)’:
/home/lenz/julia/src/llvm-late-gc-lowering.cpp:1753:41: error: no matching function for call to ‘llvm::CallInst::copyMetadata(llvm::CallInst*&)’
                 NewCall->copyMetadata(CI);
                                         ^
In file included from /home/lenz/julia/usr/include/llvm/IR/BasicBlock.h:23:0,
                 from /home/lenz/julia/usr/include/llvm/Analysis/CFG.h:18,
                 from /home/lenz/julia/src/llvm-late-gc-lowering.cpp:7:
/home/lenz/julia/usr/include/llvm/IR/Instruction.h:237:8: note: candidate: void llvm::Instruction::copyMetadata(const llvm::Instruction&, llvm::ArrayRef<unsigned int>)
   void copyMetadata(const Instruction &SrcInst,
        ^~~~~~~~~~~~
/home/lenz/julia/usr/include/llvm/IR/Instruction.h:237:8: note:   no known conversion for argument 1 from ‘llvm::CallInst*’ to ‘const llvm::Instruction&’
/home/lenz/julia/src/llvm-late-gc-lowering.cpp:1770:41: error: no matching function for call to ‘llvm::CallInst::copyMetadata(llvm::CallInst*&)’
                 NewCall->copyMetadata(CI);
                                         ^
In file included from /home/lenz/julia/usr/include/llvm/IR/BasicBlock.h:23:0,
                 from /home/lenz/julia/usr/include/llvm/Analysis/CFG.h:18,
                 from /home/lenz/julia/src/llvm-late-gc-lowering.cpp:7:
/home/lenz/julia/usr/include/llvm/IR/Instruction.h:237:8: note: candidate: void llvm::Instruction::copyMetadata(const llvm::Instruction&, llvm::ArrayRef<unsigned int>)
   void copyMetadata(const Instruction &SrcInst,
        ^~~~~~~~~~~~
/home/lenz/julia/usr/include/llvm/IR/Instruction.h:237:8: note:   no known conversion for argument 1 from ‘llvm::CallInst*’ to ‘const llvm::Instruction&’
make[1]: *** [Makefile:136: llvm-late-gc-lowering.o] Error 1
make: *** [Makefile:69: julia-src-release] Error 2



```

--- a/src/llvm-late-gc-lowering.cpp
+++ b/src/llvm-late-gc-lowering.cpp
@@ -1750,7 +1750,7 @@ bool LateLowerGCFrame::CleanupIR(Function &F, State *S) {
                 NewCall->setAttributes(attr);
 #endif
 #if JL_LLVM_VERSION >= 40000
-                NewCall->copyMetadata(CI);
+                NewCall->copyMetadata(*CI);
 #else
                 SmallVector<std::pair<unsigned, MDNode *>, 1> MDs;
                 CI->getAllMetadata(MDs);
@@ -1767,7 +1767,7 @@ bool LateLowerGCFrame::CleanupIR(Function &F, State *S) {
                 CallInst *NewCall = CallInst::Create(CI, None, CI);
                 NewCall->takeName(CI);
 #if JL_LLVM_VERSION >= 40000
-                NewCall->copyMetadata(CI);
+                NewCall->copyMetadata(*CI);
 #else
                 SmallVector<std::pair<unsigned, MDNode *>, 1> MDs;
                 CI->getAllMetadata(MDs);

